### PR TITLE
Feature 14/change filename to recipientuserid in getuploadreferralsheeturl

### PR DESCRIPTION
**Pull request by melinahernandeztheam**
## Description

GetUploadReferralSheetUrl is used to generate a signed url to upload the referral sheet to an s3. 

I would like to change the command property fileName to recipientUserId because I don't want the client to tell me the filename, instead it would be the backend who creates the filename from recipientUserId so the filename always follows the same format

## Changes

- Better naming for GetUploadReferralSheetUrl -> GetRecipientUserReferralSheetUploadUrl to reference that referral sheet is a property of RecipientUser
- Change filename to recipientUserId and save the file in /referral-sheets/referral_sheet_{recipientUserId}_{YYYY-MM-DD}
- 
- Fix a minor bug found when testing. Add check to relativesIds to ensure the id is not already on the array (This was happening when the createRelative command was called twice with the same data)

## Checks

- [ ] Project Builds
- [ ] Project passes tests and checks
- [ ] Updated documentation accordingly

## Additional information

[issue](https://github.com/TheTributeCommunity/fesbal-backend/issues/14)

### CODE REVIEW

File: src/commands/get-recipient-user-referral-sheet-upload-url.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  public constructor(readonly recipientUserId: string) {}
+
+  public static async handle(command: GetRecipientUserReferralSheetUploadUrl, register: Register): Promise<string> {
+    return UploadFileService.getUrlToUploadFile(
+      GetRecipientUserReferralSheetUploadUrl.getReferralSheetPath(command.recipientUserId)
+    )
+  }
+
+  private static getReferralSheetPath(recipientUserId: string): string {
+    const currentDate = new Date().toISOString().slice(0, 10)
```
Comment (tainguyenbui) [inline src/commands/get-recipient-user-referral-sheet-upload-url.ts:19]: I think it would be better to use a date formatter here

Comment (melinahernandeztheam) [reply src/commands/get-recipient-user-referral-sheet-upload-url.ts:19]: I don't know what do you mean, something like this?

`const currentDate  = new Date().toLocaleDateString("es-ES")); `

Comment (rdiaz82) [reply src/commands/get-recipient-user-referral-sheet-upload-url.ts:19]: The problem that I see here is that using the `slice(0,10)` we are just getting the date. Could be possible to try to upload several documents the same day?

**Review by rdiaz82 — APPROVED** [review]
LGTM, just to clarify the date format

**Review by tainguyenbui — APPROVED** [review]
LGTM

**Pull request merged**